app/controllers/application_controller.rb

```diff
@@ -1,14 +1,22 @@
 class ApplicationController < ActionController::API
+  rescue_from JWT::DecodeError, with: :error_invalid
 
   def secret_key
-    "drinkthis"
+    ENV['GOOGLE_CLIENT_SECRET']
   end
 
   def encode(payload)
     JWT.encode(payload, secret_key, 'HS256')
   end
 
   def decode(token)
-    JWT.decode(token, "drinkthis", true, {algorithm: "HS256"})[0]
+    # body = JWT.decode(token, secret_key, true, {algorithm: "HS256"})[0]
+    # HashWithIndifferentAccess.new(body)
+    response = Faraday.new(url: "https://oauth2.googleapis.com").get("/tokeninfo?id_token=#{token}")
+    JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def error_invalid
+    render json: {error: 'invalid token'}
   end
 end
```

Comment:
Does this ultimately belong in application controller or do we intend to move it out at some stage? Curious about convention here, since I've never implemented this before.

Reply:
we are removing JWT and will move the faraday into a service.